python/triton/runtime/interpreter.py

```diff
@@ -726,10 +726,12 @@ def check_tensor(self, input):
             self.check_axis(arg.shape, self.axis)
 
     def to_tensor(self, ret, dtype):
+        np_dtype = _get_np_dtype(dtype)
         if hasattr(ret, "shape") and ret.shape:
+            ret = ret.astype(np_dtype)
```

Comment:
The test added here was failing because `np.sum` upcasts to `np.int64`, then bitcast was interpreting the lower and upper 32-bits as different elements resulting in the `handle.data` tensor becoming twice the size of the triton tensor.